### PR TITLE
Favorites list

### PR DESCRIPTION
**Pull request by anandsinghparihar**
- Add to Favorites list
- Remove from a Favorites list
- Display Notification   
- Resolve lint and remove unwanted modules.

### CODE REVIEW

File: src/app/market/overview-listings/preview-listing/preview-listing.component.ts [hunk 1]
```diff
@@ -27,25 +29,34 @@ export class PreviewListingComponent implements OnInit {
...
   }
 
-  addToCart(id) {
+  addToCart(id: number) {
     this.cartService.addItem(id);
   }
 
-  addToFavorites(id) {
-    this.favoritesService.addItem(id).take(1).subscribe(res => console.log(res));
+  addToFavorites(id: number) {
```
[the remark this answers is not in the document]
Comment (anandsinghparihar) [reply src/app/market/overview-listings/preview-listing/preview-listing.component.ts]: no it was id, and no need here listing object

File: src/app/market/overview-listings/overview-listings.component.ts [hunk 2]
```diff
@@ -79,6 +80,20 @@ export class OverviewListingsComponent implements OnInit, OnDestroy {
...
   }
 
+  getFavorites() {
+    this.favoritesService.getFavorites().subscribe((favorite: Favorite[]) => {
+      if (this.pages[0] && this.pages[0].listings) {
+        favorite.forEach(obj => {
+          this.pages[0].listings.forEach(item => {
+            if (obj.listingItemId === item.id) {
+              item.favorite = true;
+            }
```
[the remark this answers is not in the document]
Comment (anandsinghparihar) [reply src/app/market/overview-listings/overview-listings.component.ts]: Sorry for a static index,  I just refactor it now it also works for all cases like pagination, search, listing etc.. 
and I am with current code no refactoring by `market-state` .

File: src/app/core/market/api/template/template.model.ts [hunk 3]
```diff
@@ -1,76 +1,94 @@
...
-    public category: Category = new Category({});
-    public createdAt: string = '';
+  public category: Category = new Category({});
+  public createdAt: string = '';
 
-    public basePrice: Amount = new Amount(0);
-    public shippingPrice: Amount = new Amount(0);
+  public basePrice: Amount = new Amount(0);
+  public shippingPrice: Amount = new Amount(0);
+  public favorite: boolean;
```
Comment (kewde) [inline src/app/core/market/api/template/template.model.ts]: Move to Listing model

File: src/app/market/overview-listings/overview-listings.component.ts [hunk 4]
```diff
@@ -79,6 +80,21 @@ export class OverviewListingsComponent implements OnInit, OnDestroy {
     this.loadPage(1);
   }
 
+  getFavorites(pageNumber: number) {
```
Comment (kewde) [inline src/app/market/overview-listings/overview-listings.component.ts]: Move the check like: isListingItemFavorited(listingItemId: number): boolean

File: src/app/market/overview-listings/overview-listings.component.ts [hunk 5]
```diff
@@ -94,26 +110,28 @@ export class OverviewListingsComponent implements OnInit, OnDestroy {
...
-          this.pushNewPage(page);
-        } else {
-          this.noMoreListings = true;
+    this.listingService.search(pageNumber, max, null, search)
+      .take(1)
+      .subscribe((listings: Array<any>) => {
+        // new page
+        const page = {
+          pageNumber: pageNumber,
+          listings: listings.map(listing => new Template(listing))
```
Comment (kewde) [inline src/app/market/overview-listings/overview-listings.component.ts]: Map to Listing model instead (my bad, used the template model initially)

File: src/app/market/overview-listings/preview-listing/preview-listing.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kewde) [reply src/app/market/overview-listings/preview-listing/preview-listing.component.ts]: Can't you just use this.listing.id ?

File: src/app/market/listing/listing.component.ts [hunk 6]
```diff
@@ -16,17 +15,17 @@ interface IDate {
 })
 export class ListingComponent implements OnInit {
 
+  pictures: any = new Array();
```
Comment (kewde) [inline src/app/market/listing/listing.component.ts]: any -> Array

File: src/app/market/listing/listing.component.ts [hunk 7]
```diff
@@ -16,17 +15,17 @@ interface IDate {
...
+  price: any;
```
Comment (kewde) [inline src/app/market/listing/listing.component.ts]: number / amount?

File: src/app/market/listing/listing.component.ts [hunk 8]
```diff
@@ -16,17 +15,17 @@ interface IDate {
...
+  date: any;
```
Comment (kewde) [inline src/app/market/listing/listing.component.ts]: Date / string?

**Comment by kewde** [comment]
Has some conflicts, looking good tho

File: src/app/market/listing/listing.component.ts (hunk 7 above)
Comment (anandsinghparihar) [reply src/app/market/listing/listing.component.ts]: https://github.com/anandsinghparihar/partgui/blob/088223d4e8c40bf847cdfe3a9db5974137e952d6/src/app/market/listing/listing.component.ts#L39L42
you set object in price.

**Comment by kewde** [comment]
Fixed merge conflicts, will probably merge this weekend

**Comment by anandsinghparihar** [comment]
@kewde This branch has no conflicts with the base branch

**Comment by anandsinghparihar** [comment]
@AllienWorks could you please check this favorite icon alignment issue 
fav icon alignment
preivew icon

**Comment by rynomster** [comment]
Loads of conflicts, also @AllienWorks still needs to fix the icons?

**Comment by HenkSwardt** [comment]
The Icon on this task has been fixed by the update that @gerlofvanek did for us to all icons

image

**Comment by rynomster** [comment]
Lots of conflicts after merging the other fixes

**Comment by rynomster** [comment]
Still a conflict :(

**Comment by anandsinghparihar** [comment]
@rynomster please merge it otherwise its again conflict  :D

**Comment by HenkSwardt** [comment]
https://github.com/particl/particl-desktop/issues/873

File: src/app/market/overview-listings/preview-listing/preview-listing.component.ts [hunk 9]
```diff
@@ -1,51 +0,0 @@
...
-  openListing() {
-    const dialog = this.dialog.open(ListingComponent, {
-      data: { listing: this.listing },
-    });
-  }
-
-  getThumbnail() {
-    if (this.listing.thumbnail) {
-      // TODO: logic for main image, taking 0 here
-      return this.listing.thumbnail.dataId;
```
Comment (kewde) [inline src/app/market/overview-listings/preview-listing/preview-listing.component.ts]: Note this..

File: src/app/market/listings/listing-item/listing-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+  openListing() {
+    const dialog = this.dialog.open(PreviewListingComponent, {
+      data: {listing: this.listing},
+    });
+  }
+
+  getThumbnail() {
+    if (this.listing.thumbnail) {
+      // TODO: logic for main image, taking 0 here
+      return 'data:image/gif;base64,' + this.listing.thumbnail.data;
```
Comment (kewde) [inline src/app/market/listings/listing-item/listing-item.component.ts]: Is not the same as this...
You seem to have been working on an older version

File: src/app/market/overview-listings/preview-listing/preview-listing.component.ts [hunk 11]
```diff
@@ -1,51 +0,0 @@
...
-    } else {
-      return './assets/images/placeholder_4-3.jpg';
-    }
-
-  }
-
-  addToCart() {
-    this.cartService.addItem(this.listing.id).subscribe();
```
Comment (kewde) [inline src/app/market/overview-listings/preview-listing/preview-listing.component.ts]: same thing here... broke adding to cart

**Pull request merged**